bullet_train-super_scaffolding/lib/scaffolding/script.rb

```diff
@@ -183,7 +197,19 @@ def check_required_options_for_attributes(scaffolding_type, attributes, child, p
       puts "Adding new fields to #{child} with '#{generation_command}'".green
     end
     puts ""
-    `#{generation_command}` unless @options["skip-migration-generation"]
+
+    unless @options["skip-migration-generation"]
+      untracked_files = has_untracked_files?(git_status) ? get_untracked_files(git_status) : []
+      generation_thread = Thread.new { `#{generation_command}` }
+      generation_thread.join # Wait for the process to finish.
+
+      newly_untracked_files = has_untracked_files?(git_status) ? get_untracked_files(git_status) : []
+      if (newly_untracked_files - untracked_files).size.zero?
```

Comment:
With the other change we could do something like this:

```suggestion
      untracked_files = untracked_files_from_git
      generation_thread = Thread.new { `#{generation_command}` }
      generation_thread.join # Wait for the process to finish.

      newly_untracked_files = untracked_files_from_git
      if (newly_untracked_files - untracked_files).none?
```